recorder/src/main/cpp/controller.cc

```diff
@@ -79,31 +56,45 @@ static void populate_recorder_info(recording::RecorderInfo& ri, const Configurat
             ri.set_local_time(now);
         });
     ri.set_recorder_version(RECORDER_VERION);
-    auto now = std::chrono::steady_clock::now();
+    auto now = Time::now();
     std::chrono::duration<double> uptime = now - start_time;
     ri.set_recorder_uptime(uptime.count());
 }
 
-static void populate_issued_work_status(recording::WorkResponse& w_res, std::uint64_t work_id, recording::WorkResponse::WorkState state, recording::WorkResponse::WorkResult result) {
+static void populate_issued_work_status(recording::WorkResponse& w_res, std::uint64_t work_id, recording::WorkResponse::WorkState state, recording::WorkResponse::WorkResult result, Time::Pt& start_tm, Time::Pt& end_tm) {
```

Comment:
Every pass-by-ref is actually a pointer, which is 64-bits wide. The enums would be 32-bit wide, less with packed. So its actually cheaper to pass it by value from register-allocation PoV in case of the two enum members, and at-par from the PoV of work_id.

When it is at par or better from size PoV, pass by reference is bad from non-inlined call PoV, because it takes additional pointer-dereference (which means it is contingent upon 2 lines being cached, not 1).